.github/workflows/build-on-pr.yml

```diff
@@ -1,20 +1,29 @@
 name: Build Docker Image
 
 on:
-  push:
-    branches: 
-      - '*'
-      - '!main'
   pull_request:
-    branches: [ '*' ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
```

Comment:
This enables GHA's to write to your registry. So you don't need to maintain the token you have in the secret.